includes/gateways/paypal/includes/class-wc-gateway-paypal-request.php

```diff
@@ -72,12 +81,90 @@ public function get_request_url( $order, $sandbox = false ) {
 	 * @return string
 	 */
 	protected function limit_length( $string, $limit = 127 ) {
-		if ( strlen( $string ) > $limit ) {
-			$string = substr( $string, 0, $limit - 3 ) . '...';
+		// As the output is to be used in http_build_query which applies URL encoding, the string needs to be
+		// cut as if it was URL-encoded, but returned non-encoded (it will be encoded by http_build_query later).
+		$url_encoded_str = rawurlencode( $string );
+
+		if ( strlen( $url_encoded_str ) > $limit ) {
+			$string = rawurldecode( substr( $url_encoded_str, 0, $limit - 3 ) . '...' );
 		}
 		return $string;
 	}
 
+	/**
+	 * Get args for paypal request, except for line item args.
+	 *
+	 * @param WC_Order $order Order object.
+	 *
+	 * @return array
+	 */
+	protected function get_non_line_item_args( $order ) {
```

Comment:
This method name is bothering me for some reason :p Can we describe what this returns, rather than what it doesn't? transaction args?

Reply:
Good point, changed to `get_transaction_args`.